.github/workflows/snapshot-ci.yml

```diff
@@ -0,0 +1,267 @@
+        working-directory: ./powsybl-openrao
+        shell: bash
+
+      #BUILD DYNAWO
+      - name: Check for DYNAWO SNAPSHOT branch
+        run: ${{ env.SCRIPTS_PATH }}/check_snapshot_branch.sh "https://github.com/powsybl/powsybl-dynawo.git" ${{ env.CORE_VERSION }}
+        shell: bash
+      - name: Checkout dynawo-sources
+        uses: actions/checkout@v4
+        with:
+          repository: powsybl/powsybl-dynawo
+          ref: ${{ env.SNAPSHOT_BRANCH }}
+          path: powsybl-dynawo
+      - name: Change core version
+        run: mvn versions:set-property -Dproperty=powsybl-core.version -DnewVersion=${{ env.CORE_VERSION}} -DgenerateBackupPoms=false
+        working-directory: ./powsybl-dynawo
+        shell: bash
+      - name: Build DYNAWO
+        run: mvn -batch-mode --no-transfer-progress clean install -DskipTests
+        working-directory: ./powsybl-dynawo
+      - name: Get DYNAWO_VERSION
+        run: echo "DYNAWO_VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)" >> $GITHUB_ENV
+        working-directory: ./powsybl-dynawo
+        shell: bash
+
+      #CHECKOUT_PYPOWSYBL_DEPENCIES
+      - name: Checkout powsybl-dependencies
+        uses: actions/checkout@v4
+        with:
+          repository: powsybl/powsybl-dependencies
+          ref: main
+          path: powsybl-dependencies
+      - name: Get DEPENDENCIES_VERSION
+        run: echo "DEPENDENCIES_VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)" >> $GITHUB_ENV
+        working-directory: ./powsybl-dependencies
+        shell: bash
+
+      #UPDATE/INSTALL_PYPOWSYBL_DEPENCIES
+      - name: Update dependencies versions
+        shell: bash
+        run: |
+          mvn versions:set-property -Dproperty=powsybl-open-loadflow.version -DnewVersion=$LOADFLOW_VERSION -DgenerateBackupPoms=false
+          mvn versions:set-property -Dproperty=powsybl-core.version -DnewVersion=$CORE_VERSION -DgenerateBackupPoms=false
+          mvn versions:set-property -Dproperty=powsybl-diagram.version -DnewVersion=$DIAGRAM_VERSION -DgenerateBackupPoms=false
+          mvn versions:set-property -Dproperty=powsybl-dynawo.version -DnewVersion=$DYNAWO_VERSION -DgenerateBackupPoms=false
+          mvn versions:set-property -Dproperty=powsybl-entsoe.version -DnewVersion=$ENTSOE_VERSION -DgenerateBackupPoms=false
+          mvn versions:set-property -Dproperty=powsybl-open-rao.version -DnewVersion=$OPENRAO_VERSION -DgenerateBackupPoms=false
+        working-directory: ./powsybl-dependencies
+
+      - name: Install powsybl-dependencies
+        run: mvn -batch-mode --no-transfer-progress clean install -DskipTests
+        working-directory: ./powsybl-dependencies
+        shell: bash
+
+      #BUILD PYPOWSYBL
+      - name: Check PYPOWSYBL SNAPSHOT branch
+        run: ${{ env.SCRIPTS_PATH }}/check_snapshot_branch.sh "https://github.com/powsybl/pypowsybl.git" ${{ env.CORE_VERSION }}
+        shell: bash
+      - name: Checkout pypowsybl
```

Comment:
We should try to have consistent titles -> Checkout pypowsybl sources

Reply:
I renamed all titles